src/openms/include/OpenMS/ANALYSIS/ID/FragmentIndex.h

```diff
@@ -132,10 +175,47 @@ namespace OpenMS
      */
     struct Fragment
     {
-      UInt32 peptide_idx; // TODO: check size in sage implementation (32bit vs 64bit)
+      UInt32 peptide_idx; // 32 bit in sage
       float fragment_mz;
     };
 
+
+    /**
+     * @brief queries exactly one peak, with a set range of potential peptides, isotope error and precursor charge Hits are transfered into the a PSM
+     * @param candidates
+     * @param peak
+     * @param candidates_range
+     * @param isotope_error
+     * @param precursor_charge
+     */
+    void queryPeak(SpectrumMatchesTopN& candidates,
+                   const Peak1D& peak,
+                   std::pair<size_t, size_t> candidates_range,
+                   int16_t isotope_error,
+                   uint16_t precursor_charge);
+
+    /**
+     * @brief queries all peaks of a spectrum. loops over all preset isotope errors. The precursor-mass-window is ignored
+     * @param spectrum experimental query-spectrum
+     * @param mz
+     * @param sms
+     * @param charge
+     */
+    void closedSearch(const MSSpectrum& spectrum, float mz, SpectrumMatchesTopN& sms, uint16_t charge);
+
+    /**@brief queries all peaks of a spectrum. ignores isotope error, applies open window
+     * @param spectrum
+     * @param precursor_mass
+     * @param sms
+     * @param charge
+     */
+    void openSearch(const MSSpectrum& spectrum, float precursor_mass, SpectrumMatchesTopN& sms, uint16_t charge);
+
+    /** @brief places the k-largest elements in the front of the input array. Inside of the k-largest elements and outside the elements are not sorted
```

Comment:
so this does not trim?
is there a reason (e.g., perfomance etc.) why no trimming is performed? if so please document or rename to something with partial sort in its name.

Reply:
It does trim, doesn't it? Anyway, I looked again at openSearch and closedSearch and I think it would be best to make one function out of it, instead of two

Reply:
Yes, I was confused because your comment talks about some implementation detail (the partial sort) that suggests that it is indeed not trimmed. If you want to document how this is done I would propose doing it in the cpp file.
A comment that hides this implementation detail could be something like: @brief keeps the top k candidates (according to number of matched fragments)